numba/npyufunc/ufuncbuilder.py

```diff
@@ -84,6 +144,17 @@ def parse_identity(self, identity):
             raise ValueError("Invalid identity value %r" % (identity,))
         return identity
 
+    def add(self, sig=None, argtypes=None, restype=None):
+        if hasattr(self, 'targetoptions'):
```

Comment:
I moved the add() method up to the common base class.  While not strictly holding to OO-methodology, this branch is to handle where different subclasses keep the compiler flags.